### PR TITLE
LSIF: update user documentation

### LINKED ISSUE #6283: LSIF: workflow validation (internal)
The top priority in the [roadmap](https://docs.google.com/document/d/1cBsE9801DcBF9chZyMnxRdolqM_1c2pPyGQz15QAvYI/edit#heading=h.dimwsc9ccmwq) for the GA announcement of LSIF is to have 2 Sourcegraphers set it up and try it out for a test run 🚀 The goal is to identify rough spots, wrong documentation, bugs, and see if we can make the docs better.

@kzh @attfarhan You are the chosen ones! Expect a calendar invitation to a 30-minute screen share with the @sourcegraph/code-intel team so we can note any issues as they happen and improve the workflow 😃 

No need to prepare - we'll go over everything during the meeting. This is roughly what you'll do:

## Test plan 1 @kzh

Follow https://docs.sourcegraph.com/user/code_intelligence/lsif to get precise code intel on Sourcegraph.com for the repository https://github.com/sourcegraph/lsif-java

## Test plan 2 @attfarhan 

Follow https://docs.sourcegraph.com/user/code_intelligence/lsif to get precise code intel on on your own Sourcegraph 3.9.3 instance for the repository https://github.com/sourcegraph/sourcegraph, only for the `cmd/` and `shared/` directories.

### PR DESCRIPTION
**Pull request by efritz**
Closes #6283

### CODE REVIEW

File: doc/user/code_intelligence/lsif_in_ci.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+From the command-line:
+
+```
+$ src \
+  -endpoint=https://sourcegraph.example.com \
+  lsif upload \
+  -github-token=abc... (only needed when uploading to Sourcegraph.com) \
+  -repo=github.com/<user>/<reponame> \
+  -commit=$(git rev-parse HEAD | tr -d "\n") \
+  -root=<project directory with a trailing slash> (omit when the project root is the same as the repository root)
```
Comment (felixfbecker) [inline doc/user/code_intelligence/lsif_in_ci.md]: Can't the CLI add the trailing slash if it's missing? Repo root or cwd?

File: doc/user/code_intelligence/lsif_in_ci.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+From the command-line:
+
+```
+$ src \
+  -endpoint=https://sourcegraph.example.com \
+  lsif upload \
+  -github-token=abc... (only needed when uploading to Sourcegraph.com) \
+  -repo=github.com/<user>/<reponame> \
+  -commit=$(git rev-parse HEAD | tr -d "\n") \
```
Comment (felixfbecker) [inline doc/user/code_intelligence/lsif_in_ci.md]: Can the CLI trim the newline, or even default to running this command?

File: doc/user/code_intelligence/lsif_on_github.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  - push
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Generate LSIF Data
+        uses: sourcegraph/lsif-go-action@master
+        with:
+          verbose: "true"
```
Comment (felixfbecker) [inline doc/user/code_intelligence/lsif_on_github.md]: Why is the boolean quoted?

File: doc/user/code_intelligence/lsif_in_ci.md (hunk 1 above)
Comment (efritz) [reply doc/user/code_intelligence/lsif_in_ci.md]: We might want to accept missing slashes on the server and add it there (if it's != '' and doesn't end with /, append one)

File: doc/user/code_intelligence/lsif_in_ci.md (hunk 2 above)
Comment (efritz) [reply doc/user/code_intelligence/lsif_in_ci.md]: This command doesn't need to be run in the tree that generated the dump, though. I'm not sure that's an intuitive default.

Comment (efritz) [reply doc/user/code_intelligence/lsif_in_ci.md]: Trimming the newline would be fine to do, though.

File: doc/user/code_intelligence/lsif_on_github.md (hunk 3 above)
Comment (efritz) [reply doc/user/code_intelligence/lsif_on_github.md]: I think all params must be strings in actions.

File: doc/user/code_intelligence/lsif_on_github.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Upload LSIF data
+        uses: sourcegraph/lsif-upload-action@master
+        with:
+          public_repo_github_token: ${{ secrets.PUBLIC_REPO_GITHUB_TOKEN }}
+```
+
+Once that workflow is committed to your repository, you will start to see LSIF workflows in the Actions tab of your repository (e.g. https://github.com/sourcegraph/sourcegraph/actions).
+
+![img/workflow.png](img/workflow.png)
```
Comment (efritz) [inline doc/user/code_intelligence/lsif_on_github.md]: Is this asset missing?

File: doc/user/code_intelligence/lsif_in_ci.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# LSIF in continuous integration
+
+## LSIF indexers
+
+An LSIF indexer is a command line tool that analyzes your project's source code and generates a file in LSIF format containing all the definitions, references, and hover documentation in your project. That LSIF file is later uploaded to Sourcegraph to provide code intelligence.
+
+## Generating and uploading LSIF in CI
+
+1. Install the [Sourcegraph CLI (`src`)](https://github.com/sourcegraph/src-cli) for uploading LSIF data on your CI machines.
+1. Go to https://lsif.dev to find an LSIF indexer for your language and install the command-line tool on your CI machines.
```
Comment (efritz) [inline doc/user/code_intelligence/lsif_in_ci.md]: Do we want to just put the usage documentation in here or make sure that the usage is clear in the readme of each project (might be hard if we don't own it)?

File: doc/user/code_intelligence/lsif_in_ci.md (hunk 1 above)
Comment (efritz) [reply doc/user/code_intelligence/lsif_in_ci.md]: Done in https://github.com/sourcegraph/sourcegraph/pull/6416.

Comment (chrismwendt) [reply doc/user/code_intelligence/lsif_in_ci.md]: Making missing slashes optional in https://github.com/sourcegraph/sourcegraph/pull/6419

File: doc/user/code_intelligence/lsif_on_github.md (hunk 4 above)
Comment (chrismwendt) [reply doc/user/code_intelligence/lsif_on_github.md]: Added missing asset.

File: doc/user/code_intelligence/lsif_in_ci.md (hunk 5 above)
Comment (chrismwendt) [reply doc/user/code_intelligence/lsif_in_ci.md]: I could see us going either way. There wouldn't be much maintenance overhead to keeping a single-line command up to date, and would help us apply conventions on paths (e.g. `dump.lsif` should go in the directory it's associated with). For now, let's rely on the respective READMEs and update them to be user friendly.

**Pull request merged**